src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/GrapplingHook.java

```diff
@@ -22,19 +22,22 @@
 
 public class GrapplingHook extends SimpleSlimefunItem<ItemUseHandler> {
 
+    private final ItemSetting<Boolean> stayininvonuse = new ItemSetting<>("stay-in-inv-on-use", false);
```

Comment:
Please follow the naming convention for fields, this should be in "pascal case", so "stayInInvOnUse".
Other than that I think the name is a little confusing, I think something like `consumeOnUse` (which would actually be the opposite here) would make more sense and is much more user-friendly.